src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2331,19 +2331,18 @@ private bool ConsumeCommentMultiSegment()
             ReadOnlySpan<byte> localBuffer = _buffer.Slice(_consumed + 1);
             int leftOver = 2;
 
-            SequencePosition start = new SequencePosition(_currentPosition.GetObject(), _currentPosition.GetInteger() + _consumed);
             if (localBuffer.Length == 0)
             {
                 if (IsLastSpan)
                 {
-                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, JsonConstants.Slash);
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.UnexpectedEndOfDataWithinStartOfComment);
```

Comment:
Update to throw the same exception in `ConsumeComment`
https://github.com/dotnet/corefx/blob/aa6b47f9581eb7052b14a0a94db231938c12e599/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L2192

Reply:
Also, why is this `UnexpectedEndOfDataWithinStartOfComment` when the one below `ExpectedComment`?

I would just use `ExpectedComment` (i.e. `Expected a comment, but instead reached end of data.`) in both places. I don't see why specifying that we expected a delimiter makes sense here (and if you think it does make sense, why is the case below different?).

Reply:
The line below and its second instance, mentions a `Slash` but it could actually be a `/` or a `*`. I felt that intention behind original exception message (with `Slash`) was to be informative but fell short of correctness in terms of `/` or `*` as second character.

https://github.com/dotnet/corefx/blob/87fdc75beacc76e02c8e572ae3b0613b418200eb/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L2339

Hence the new exception which conveys the same thing i.e. in middle of delimiter, without saying which character was expected.

> Also, why is this UnexpectedEndOfDataWithinStartOfComment when the one below ExpectedComment?

`ExpectedComment` has been used in second place because at that program point `//` or `/*` has been parsed and we are really expecting comment text there.

> Update to throw the same exception in ConsumeComment

Agreed. Doing the change.

Reply:
I don't necessarily see the user benefiting much from knowing precisely why the error occurred when we are in the comment cases. To me, all these are equivalent "error" states as far as the user's concerned:
1) `/|` - early end of data - `Expected a comment, but instead reached end of data.`
2) `//|` - early end of data - `Expected a comment, but instead reached end of data.`
3) `//h|` - early end of data - `Expected a comment, but instead reached end of data.`

Why would the user need to know that they are in (2)/(3) instead of (1) and how would their action differ because of it.
Currently, we return `Expected comment delimiter, but instead reached end of data.` for that, but why?

If we are really want different exception messages, let's be consistent and throw the right one at the right time for all inputs (single-segment and multi-segment) - see https://github.com/dotnet/corefx/pull/35934/files#r281055434

cc @bartonjs, do you have any thoughts on this?

Reply:
I don't see the value.  I'd call all of (1)-(3) `Unexpected end of data.` or, if I was feeling like having a lot of messages, `Unexpected end of data while reading a comment.`

Reply:
@WinCPP - let me know when you plan to address this final feedback + resolve the merge conflict. Thanks for your effort dealing with all the subtleties.